internal/pkg/util/fs/overlay/overlay_item_linux.go

```diff
@@ -0,0 +1,297 @@
+		return fmt.Errorf("error accessing directory %s: %s", i.DirToMount, err)
+	}
+
+	sylog.Debugf("Performing identity bind-mount of %q", i.DirToMount)
+	if err = syscall.Mount(i.DirToMount, i.DirToMount, "", syscall.MS_BIND, ""); err != nil {
+		return fmt.Errorf("failed to bind %s: %s", i.DirToMount, err)
+	}
+
+	// Best effort to cleanup mount
+	defer func() {
+		if err != nil {
+			sylog.Debugf("Encountered error with current OverlaySet; attempting to unmount %q", i.DirToMount)
+			syscall.Unmount(i.DirToMount, syscall.MNT_DETACH)
+		}
+	}()
+
+	// Try to perform remount
+	sylog.Debugf("Performing remount of %q", i.DirToMount)
+	if err = syscall.Mount("", i.DirToMount, "", syscall.MS_REMOUNT|syscall.MS_BIND, ""); err != nil {
+		return fmt.Errorf("failed to remount %s: %s", i.DirToMount, err)
+	}
+
+	return nil
+}
+
+// mountWithFuse mounts an image to a temporary directory using a specified fuse
+// tool. It also verifies that fusermount is present before performing the
+// mount.
+func (i *Item) mountWithFuse(fuseMountTool string) error {
+	var err error
+	fuseMountCmd, err := bin.FindBin(fuseMountTool)
+	if err != nil {
+		return fmt.Errorf("use of image %q as overlay requires %s to be installed: %s", i.BarePath, fuseMountTool, err)
+	}
+
+	// Even though fusermount is not needed for this step, we shouldn't perform
+	// the mount unless we have the necessary tools to eventually unmount it
+	_, err = bin.FindBin("fusermount")
+	if err != nil {
+		return fmt.Errorf("use of image %q as overlay requires fusermount to be installed: %s", i.BarePath, err)
+	}
+
+	// Obtain parent directory in which to create overlay-related mount
+	// directories. See https://github.com/apptainer/singularity/pull/5575 for
+	// related discussion.
+	parentDir, err := i.GetSecureParentDir()
+	if err != nil {
+		return fmt.Errorf("error while trying to create parent dir for overlay %q: %s", i.BarePath, err)
+	}
+	fuseMountDir, err := os.MkdirTemp(parentDir, "overlay-mountpoint-")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary dir %q for overlay %q: %s", fuseMountDir, i.BarePath, err)
+	}
+
+	// Best effort to cleanup temporary dir
+	defer func() {
+		if err != nil {
+			sylog.Debugf("Encountered error with current OverlaySet; attempting to remove %q", fuseMountDir)
+			os.Remove(fuseMountDir)
+		}
+	}()
+
+	execCmd := exec.Command(fuseMountCmd, i.BarePath, fuseMountDir)
+	execCmd.Stderr = os.Stderr
+	_, err = execCmd.Output()
+	if err != nil {
+		return fmt.Errorf("encountered error while trying to mount image %q as overlay at %s: %s", i.BarePath, fuseMountDir, err)
+	}
+	i.DirToMount = fuseMountDir
+
+	return nil
+}
+
+// Unmount performs the necessary steps to unmount an individual OverlayItem.
+// Note that this method does not unmount the overlay itself. That happens in
+// OverlaySet.Unmount().
+func (i Item) Unmount() error {
+	switch i.Type {
+	case image.SANDBOX:
+		return i.unmountDir()
+
+	case image.SQUASHFS:
+		fallthrough
```

Comment:
Is there a reason you aren't calling unmountFuse here?

Reply:
Yeah sorry about this one... it's obvious today... I must have had brain fade on my first run through :-)